scrapyrt/core.py

```diff
@@ -175,9 +175,11 @@ def spider_idle(self, spider):
             assert callable(callback), 'Invalid callback'
             self.request = self.request.replace(callback=callback)
 
-            errback = getattr(self.crawler.spider, self.errback_name)
-            assert callable(errback), 'Invalid errback'
-            self.request = self.request.replace(errback=errback)
+
+            if self.errback_name:
+                errback = getattr(self.crawler.spider, self.errback_name)
+                assert callable(errback), 'Invalid errback'
```

Comment:
the problem with this here is that on bad parameters for errback it will fail silently on API side and write errors to logs. If errback is not callable or spider has no "errback_name". Possibly we could handle this better by adding & validating errback on crawlerManager init and then wraping line 254 in run_crawl to throw 400 HTTP error if errback provided but not callable. 

The line
```python
getattr(self.crawler.spider, self.errback_name)
```

can also fail if errback is not an attribute of spider so in this case I would also try to throw 400 status code as this would be user error. 

I'm not really sure if we should keep backward compat for behavior which seems to be an error. If parse should not be default errback then having it as default is not helping people. We can add warning to new release about this.

It also needs a unit test. So please add unit test. 

Thank you for providing this fix. I'm happy to merge it and release it this week.